src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+                    case "MultiThreaded":
+                        return "/MT"; // defines macro "_MT"
+                    case "MultiThreadedDebug":
+                        return "/MTd"; // defines macros "_MT" and "_DEBUG"
+                    case "MultiThreadedDLL":
+                    case "MultiThreadedDll":
+                        return "/MD"; // defines macros "_MT" and "_DLL"
+                    case "MultiThreadedDebugDLL":
+                    case "MultiThreadedDebugDll":
+                        return "/MDd"; // defines macros "_MT", "_DLL" and "_DEBUG"
+                }
+            }
+
+            internal static string ConvertEnableEnhancedInstructionSet(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported EnableEnhancedInstructionSet: " + value);
+                    case "NotSet":
+                        return "";
+                    case "AdvancedVectorExtensions":
+                        return "/arch:AVX";
+                    case "AdvancedVectorExtensions2":
+                        return "/arch:AVX2";
+                    case "StreamingSIMDExtensions":
+                        return "/arch:SSE";
+                    case "StreamingSIMDExtensions2":
+                        return "/arch:SSE2";
+                    case "NoExtensions":
+                        return "/arch:IA32";
+                }
+            }
+
+            internal static string ConvertExceptionHandling(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported ExceptionHandling: " + value);
+                    case "false":
+                        return "";
+                    // all options below define macro "_CPPUNWIND":
+                    case "Async":
+                        return "/EHa";
+                    case "Sync":
+                        return "/EHsc";
+                    case "SyncCThrow":
+                        return "/EHs";
+                }
+            }
+
+            internal static string ConvertBasicRuntimeChecks(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported BasicRuntimeChecks: " + value);
+                    case "Default":
+                        return "";
+                    // all options below define macro "__MSVC_RUNTIME_CHECKS":
+                    case "StackFrameRuntimeCheck":
+                        return "/RTCs";
+                    case "UninitializedLocalUsageCheck":
+                        return "/RTCu";
+                    case "EnableFastChecks":
+                        return "/RTC1";
+                }
+            }
+        }
+
+        private static void AddList(List<string> cmd, string[] values)
```

Comment:
@Evangelink What is the meaning of "range of command line arguments"? It is literally a list of command-line arguments in the sense of an English meaning of word "list". What's the added value in naming it differently?